core-web/apps/dotcms-ui/src/app/portlets/shared/dot-content-types-edit/components/dot-binary-settings/dot-binary-settings.component.ts

```diff
@@ -99,77 +87,64 @@ export class DotBinarySettingsComponent implements OnInit, OnChanges {
+                allowCodeWrite: false,
+                allowFileNameEdit: false
+            })
         });
 
-        this.form.valueChanges.subscribe(() => {
+        this.form.valueChanges.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(() => {
             this.valid.emit(this.form.valid);
         });
 
-        this.fieldVariablesService
-            .load(this.field)
-            .subscribe((fieldVariables: DotFieldVariable[]) => {
+        this.fieldVariablesService.load(this.field).subscribe({
+            next: (fieldVariables: DotFieldVariable[]) => {
                 fieldVariables.forEach((variable) => {
                     const { key, value } = variable;
-
-                    if (key === 'accept') {
-                        this.settingsMap.accept.variable = variable;
-                        this.form.get(key)?.setValue(value);
-                    } else if (key === 'systemOptions') {
-                        this.settingsMap.systemOptions.variable = variable;
+                    const control = this.form.get(key);
+                    if (control instanceof FormGroup) {
                         const systemOptions = JSON.parse(value);
 
                         this.systemOptions.forEach(({ key }) => {
-                            this.form.get(key)?.setValue(systemOptions[key]);
+                            control.get(key)?.setValue(systemOptions[key]);
                         });
+                    } else {
+                        control.setValue(value);
                     }
                 });
-            });
+            }
+        });
     }
 
     saveSettings(): void {
-        const updateActions = this.settings.map(({ variable, key }) => {
-            let fieldVariable: DotFieldVariable;
-            let value: string | boolean;
-
-            if (key === 'accept') {
-                value = this.form.get(key).value;
-                fieldVariable = {
-                    ...variable,
-                    key,
-                    value
-                };
-            } else if (key === 'systemOptions') {
-                value = JSON.stringify(
-                    this.systemOptions.reduce((acc, { key }) => {
-                        acc[key] = this.form.get(key).value;
-
-                        return acc;
-                    }, {})
-                );
-
-                fieldVariable = {
-                    ...variable,
-                    key,
-                    value
-                };
-            }
+        const updateActions = Object.keys(this.form.controls).map((key) => {
+            const control = this.form.get(key);
+            const value =
+                control instanceof FormGroup
+                    ? JSON.stringify(
+                          this.systemOptions.reduce((acc, { key }) => {
+                              acc[key] = control.get(key).value;
+
+                              return acc;
+                          }, {})
```

Comment:
Do we need this? If a control is a nested `FormGroup`, `control.value` will provide an object with a key (control name) and a value for each control.

Reply:
Right!
Changed!